components/Layout/HeadMeta.tsx

```diff
@@ -0,0 +1,31 @@
+import Head from 'next/head';
+import { Meta } from '../../lib/types';
+
+type MetaProps = {
+  meta: Meta;
+};
+
+export const HeadMeta = ({ meta }: MetaProps) => {
+  const {
+    title = 'The Best Recipes - Yummy Recipes!',
+    description = 'These are yummy recipes.',
+    keywords = 'food, appetizers, desserts, drinks',
+    type = 'article',
+    image,
+    url,
+  } = meta;
+
+  return (
+    <Head>
+      <title key="title">{title}</title>
+      {url && <link key="canonical" rel="canonical" href={url} />}
```

Comment:
Q: Is there a case when we don't have an url?

Reply:
There could be cases when you might not want to set a canonical, so I added that.